src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/germline/IntegerCopyNumberExpectationsCalculator.java

```diff
@@ -0,0 +1,342 @@
+                    " is enabled");
+        }
+
+        /* calculate E[log(c)] and E[log(c)^2] */
+        final double[] hiddenStatesLogCopyRatios = hiddenStates.stream()
+                .mapToDouble(IntegerCopyNumberState::getLogCopyNumber).toArray();
+        final double[] hiddenStatesLogCopyRatiosSquared = hiddenStates.stream()
+                .mapToDouble(IntegerCopyNumberState::getLogCopyNumberSquared).toArray();
+
+        /* exponentiate log copy number posteriors on all targets */
+        final List<double[]> hiddenStateProbabilities = IntStream.range(0, targetList.size())
+                .mapToObj(ti -> genotypeSpecificHMM.hiddenStates().stream()
+                        .mapToDouble(s -> FastMath.exp(result.logProbability(ti, s)))
+                        .toArray())
+                .collect(Collectors.toList());
+
+        if (CHECK_FOR_NANS) {
+            final int[] badTargets = IntStream.range(0, targetList.size())
+                    .filter(ti -> Arrays.stream(hiddenStateProbabilities.get(ti))
+                            .anyMatch(p -> Double.isNaN(p) || Double.isInfinite(p))).toArray();
+            if (badTargets.length > 0) {
+                throw new RuntimeException("Some of the copy ratio posterior probabilities are ill-defined; targets: " +
+                        Arrays.stream(badTargets).mapToObj(String::valueOf).collect(Collectors.joining(", ", "[", "]")));
+            }
+        }
+
+        /* calculate copy ratio posterior mean and variance */
+        final double[] logCopyRatioPosteriorMeans = hiddenStateProbabilities.stream()
+                .mapToDouble(pdf -> calculateMeanDiscreteStates(includedHiddenStatesIndices, pdf,
+                        hiddenStatesLogCopyRatios))
+                .toArray();
+        final double[] logCopyRatioPosteriorVariances = IntStream.range(0, targetList.size())
+                .mapToDouble(ti -> calculateMeanDiscreteStates(includedHiddenStatesIndices,
+                        hiddenStateProbabilities.get(ti), hiddenStatesLogCopyRatiosSquared)
+                            - FastMath.pow(logCopyRatioPosteriorMeans[ti], 2))
+                .toArray();
+
+        if (CHECK_FOR_NANS) {
+            final int[] badTargets = IntStream.range(0, targetList.size())
+                    .filter(ti -> Double.isNaN(logCopyRatioPosteriorMeans[ti]) ||
+                            Double.isNaN(logCopyRatioPosteriorVariances[ti])).toArray();
+            if (badTargets.length > 0) {
+                throw new RuntimeException("Some of the copy ratio posterior expectations are ill-defined; targets: " +
+                        Arrays.stream(badTargets).mapToObj(String::valueOf)
+                                .collect(Collectors.joining(", ", "[", "]")));
+            }
+        }
+
+        return new CopyRatioExpectations(logCopyRatioPosteriorMeans, logCopyRatioPosteriorVariances);
+    }
+
+    @Override
+    public CopyRatioExpectations getCopyRatioPriorExpectations(
```

Comment:
and there are lots of little differences... I'm not sure refactoring really helps.